doc/api/events.markdown

```diff
@@ -377,6 +377,42 @@ listener array. If any single listener has been added multiple times to the
 listener array for the specified `event`, then `removeListener` must be called
 multiple times to remove each instance.
 
+Note that once an event has been emitted, all listeners attached to it at the
+time of emitting will be called in order. This implies that any `removeListener`
+call *after* emitting and *before* the last listener finishes execution will
+not affect the current listener array. Subsequent events will behave as expected.
```

Comment:
Perhaps "will not affect the current listener array" is a bit misleading. To me it sounds like it's saying that listeners won't be removed at all, when in fact they actually *are* removed but it just doesn't affect the list of listeners used by `emit()`. Maybe it could instead read something like:

```
Note that once an event has been emitted, all listeners attached to it at the
time of emitting will be called in order. Any calls to `removeListener()` or
`removeAllListeners()` for this event will not affect an `emit()` in progress.
```